javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Validator.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.fwoa;
+
+import java.io.InputStream;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+import org.hl7.fhir.common.hapi.validation.support.CommonCodeSystemsTerminologyService;
+import org.hl7.fhir.common.hapi.validation.support.InMemoryTerminologyServerValidationSupport;
+import org.hl7.fhir.common.hapi.validation.support.PrePopulatedValidationSupport;
+import org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain;
+import org.hl7.fhir.common.hapi.validation.validator.FhirInstanceValidator;
+import org.hl7.fhir.r4.model.CodeSystem;
+import org.hl7.fhir.r4.model.StructureDefinition;
+import org.hl7.fhir.r4.model.ValueSet;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.support.DefaultProfileValidationSupport;
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.parser.StrictErrorHandler;
+import ca.uhn.fhir.rest.server.exceptions.InvalidRequestException;
+import ca.uhn.fhir.validation.FhirValidator;
+import ca.uhn.fhir.validation.ValidationResult;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.Resource;
+import io.github.classgraph.ResourceList;
+import io.github.classgraph.ScanResult;
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.fwoa.models.IgFile;
+import software.amazon.fwoa.models.IgIndex;
+
+/**
+ * This class is a wrapper around the HAPI FhirValidator.
+ * The FhirValidator is built using default settings and the available implementation guides are loaded into it.
+ */
+@Slf4j
+public class Validator {
+    private static final Gson GSON = new Gson();
+    public static final String IMPLEMENTATION_GUIDES_FOLDER = "implementationGuides";
+
+    private final FhirValidator validator;
+
+    public Validator() {
+        // To learn more about the different ways to configure FhirInstanceValidator see: https://hapifhir.io/hapi-fhir/docs/validation/validation_support_modules.html
+        FhirContext ctx = FhirContext.forR4();
```

Comment:
Pass in version to have R4 or STU3?

Reply:
I'll add support for STU3 later. 

Right now there is a very silly bug that prevents loading ValueSets from IGs into the validator
https://chat.fhir.org/#narrow/stream/179167-hapi/topic/PrePopulatedValidationSupport/near/217423371

Reply:
https://github.com/hapifhir/hapi-fhir/issues/2432